From: Boian Bonev <bbonev@ipacct.com>
Date: Tue Oct 17 01:23:35 UTC 2023
Forwarded: 629d80290c34b3a6a2e6f6400d8e277597547c93
Subject: Fix ASCII graph problem in R, W & RW modes

---
--- a/src/view_curses.c
+++ b/src/view_curses.c
@@ -969,21 +969,21 @@ static inline void view_curses(struct xx
 								v1=value2scale(s->readhist[j*2],maxvisible);
 								v2=value2scale(s->readhist[j*2+gi],maxvisible);
 							} else
-								v1=value2scale(s->readhist[j*2],maxvisible);
+								v1=value2scale(s->readhist[j],maxvisible);
 							break;
 						case E_GR_W:
 							if (has_unicode&&config.f.unicode) {
 								v1=value2scale(s->writehist[j*2],maxvisible);
 								v2=value2scale(s->writehist[j*2+gi],maxvisible);
 							} else
-								v1=value2scale(s->writehist[j*2],maxvisible);
+								v1=value2scale(s->writehist[j],maxvisible);
 							break;
 						case E_GR_RW:
 							if (has_unicode&&config.f.unicode) {
 								v1=value2scale(s->readhist[j*2]+s->writehist[j*2],maxvisible);
 								v2=value2scale(s->readhist[j*2+gi]+s->writehist[j*2+gi],maxvisible);
 							} else
-								v1=value2scale(s->readhist[j*2]+s->writehist[j*2],maxvisible);
+								v1=value2scale(s->readhist[j]+s->writehist[j],maxvisible);
 							break;
 						case E_GR_SW:
 							if (has_unicode&&config.f.unicode) {
